src/uthenticode.cpp

```diff
@@ -140,16 +140,26 @@ std::ostream &operator<<(std::ostream &os, checksum_kind kind) {
   }
 }
 
+static inline bool name_to_string(std::string &strname, X509_NAME *name, unsigned long flags) {
+  std::unique_ptr<BIO, decltype(&BIO_free)> name_bio(BIO_new(BIO_s_mem()), BIO_free);
+  if (-1 != X509_NAME_print_ex(name_bio.get(), name, 0, (flags) & ~(ASN1_STRFLGS_ESC_MSB))) {
+    char *data = nullptr;
+    auto len = BIO_get_mem_data(name_bio.get(), &data);
+    if (data && len) {
+      strname = std::string(data, len);
+      return true;
+    }
+  }
+  return false;
+}
+
 Certificate::Certificate(X509 *cert) {
-  auto subject = impl::OpenSSL_ptr(X509_NAME_oneline(X509_get_subject_name(cert), nullptr, 0),
-                                   impl::OpenSSL_free);
-  auto issuer = impl::OpenSSL_ptr(X509_NAME_oneline(X509_get_issuer_name(cert), nullptr, 0),
-                                  impl::OpenSSL_free);
+  constexpr auto xn_flags = default_xn_flags;
+  (void) name_to_string(issuer_, X509_get_issuer_name(cert), xn_flags);
+  (void) name_to_string(subject_, X509_get_subject_name(cert), xn_flags);
```

Comment:
Just making sure I understand: you've put `(void)` here to tell the compiler to ignore the return type, right? If so assigning to `std::ignore` is probably a little more idiomatic, or just dropping the return type from `name_to_string` entirely.